jormungandr/src/network/client.rs

```diff
@@ -70,13 +72,21 @@ where
                     .map(move |service| (service, peer_comms, block_req))
             })
             .and_then(move |(mut service, mut peer_comms, block_req)| {
-                let gossip_req = service.gossip_subscription(peer_comms.subscribe_to_gossip());
-                block_req
-                    .join(gossip_req)
-                    .map(move |(block_res, gossip_res)| {
-                        (service, peer_comms, block_res, gossip_res)
-                    })
+                let content_req = service.content_subscription(peer_comms.subscribe_to_fragments());
+                service
+                    .ready()
+                    .map(move |service| (service, peer_comms, block_req, content_req))
             })
+            .and_then(
+                move |(mut service, mut peer_comms, block_req, content_req)| {
+                    let gossip_req = service.gossip_subscription(peer_comms.subscribe_to_gossip());
+                    block_req
+                        .join(gossip_req)
```

Comment:
`join` is going to yield result only if both `gossip_req` and `block_req` are received, right? Is this correct, should we process them grouped and only if both requests were received?

Reply:
This resolves successfully when all requests succeed.

I also need to plug in the content request. I'll push a new commit shortly.